### ISSUE TITLE
Permeate parallel usage

### ISSUE THREAD

**Issue by AladW** [AladW/aurutils#391]
Right now `parallel` is an optional dependency for `aur-srcver`, but I think there is much to gain in using it in other scripts as well. In particular it is vastly more flexible than `xargs`, e.g. `{}` can be expanded to several arguments, and can easily be used in combination with `wget -i -` for threaded downloads.

I had imported `xe` as a submodule with this in mind, but `parallel` already has all its features, a longer trackrecord and does not require additional compiling. It is also in the community repo (only depending on `perl` and `procps-ng` which are installed on 100% of Arch systems according to pkgstats).

The only (practical) downside compared to `xe` I can think of is that by default, a `.parallel` dotfile is created. This can however be moved elsewhere with `PARALLEL_HOME` introduced with https://git.savannah.gnu.org/cgit/parallel.git/commit/?id=685018f532f4e2d24b84eb28d5de3d759f0d1af1

----
In summary:
* **xe** - Pros
  * Has basic `parallel` abilities including line buffering of output
  * Better interface than `xargs`, such as `-I{}` with multiple arguments and default new-line separation
  * Good performance
* **xe** - Cons
  * Not in the repos, either dependency on AUR package or import as submodule
  * Relatively new project, will probably find bugs during usage
* **parallel** - Pros
  * Highly flexible with many options for transforming command lines
  * Long-established project
* **parallel** - Cons
  * Requires use of `--will-cite` to not spam an ad on donating 10k EUR to the author (alternative: single use of `printf 'will cite\n' | parallel --bibtex`)
  * Minimal startup time of 0.10 seconds
  * Writes ~/.parallel unless `PARALLEL_HOME` is set

**Comment by rafasc** [comment]
Based that alone, I don't see a strong argument to use `xe`.

**Comment by AladW** [comment]
There's one other small thing against parallel, in that it takes 0.10 seconds to start. I guess this is negligible.

**Comment by AladW** [comment]
Also having to use `--will-cite` in every invocation gets pretty annoying. I guess you could instead mention users to run:
```
printf 'will cite\n' | parallel --bibtex
```

**Comment by AladW** [comment]
I updated the original post with a summary of pro/cons. I tend to go towards `xe`, but it's annoying for it to be an AUR package (and it would hardly justified bringing it to the repos with 0 votes and `parallel` as competitor).

**Comment by maximbaz** [comment]
Personally I don't see either mentioned cons of parallel being an issue, to be honest.

* Don't use `--will-cite`, let the notice be printed on stderr, and mention somewhere that it is possible to silence it by running `parallel --bibtex` (so that people can read the notice and consciously type "will cite").
* 0.1s feels negligible
* Writes to `~/.parallel`... so what 🙂

**Comment by AladW** [comment]
All right, I'll file a PR then. :slightly_smiling_face: 
```
6 files changed, 10 insertions(+), 36 deletions(-)
```
:grin:

**Comment by rafasc** [comment]
Some more thoughts on the issue with `--citation` here. https://news.ycombinator.com/item?id=12943150

And and they seem to avoid giving clarification on what the implications on the contradicting licence terms. https://lists.gnu.org/archive/html/parallel/2016-03/msg00024.html

**Comment by AladW** [comment]
Hm, https://www.gnu.org/licenses/gpl-faq.en.html#RequireCitation makes things pretty clear yet the parallel authors stick their head in the sand.

There's some other things that bother me about the authors to be honest. Whenever a new tool for command-line processing shows up, `parallel(1)` does all it can to destroy this tool and claim the superiority of *parallel*. :frowning_face: 

This is why we can't have nice things...

----
Note: most of the changes in #394 can be applied verbatim to `xe` should we decide against `parallel` after all.

**Comment by AladW** [comment]
As per discussion in #394, something like this might be suitable:
```
lib/aur-jobs
----
readonly argv0=jobs
readonly XDG_CONFIG_HOME=${XDG_CONFIG_HOME:-$HOME/.config}

# Isolate parallel directory for aurutils usage.
export PARALLEL_HOME=$XDG_CONFIG_HOME/aurutils/parallel

# By default, parallel displays a request to either pay 10000 EUR to the author 
# or make a citation about usage of parallel. Due to GPL (under which parallel is licensed)
# explicitely disallowing such behavior, this notice is disabled unconditionally for aurutils usage.
# See https://github.com/AladW/aurutils/issues/391 for details.
exec parallel --will-cite "$@"
```

**Comment by Earnestly** [comment]
> Writes to ~/.parallel... so what 🙂
ಠ_ಠ

**Comment by AladW** [comment]
Fixed by #391 / `aur-jobs`
